MetaMask/metamask-extension
# Get infuraBlocked from network state

Currently, infuraBlocked is managed via the PreferencesController through the `NetworkController:infuraIsBlocked` and `NetworkController:infuraIsUnblocked` events. As part of multichain work, we want to move away from a globally selected network. This PR removes `state.metamask.infuraBlocked` and checks `state.metamask.networksMetadata` for network status instead. This still checks the currently the state for the selected network only though.

* Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1033
* See: https://github.com/MetaMask/core/pull/1557

## Explanation



## Screenshots/Screencaps



### Before



### After



## Manual Testing Steps

Not super easy to test manually as you will need to use a VPN to connect from a location where infura is blocked or some how catch and modify the initial wallet rpc request to return a blocked error. You should see the infura blocked popup on the main wallet homepage when you have selected an infura RPC while connecting from a blocked location



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
